packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -300,10 +300,12 @@ class RawAutocomplete<T extends Object> extends StatefulWidget {
 }
 
 class _RawAutocompleteState<T extends Object> extends State<RawAutocomplete<T>> {
-  final GlobalKey _fieldKey = GlobalKey();
   final LayerLink _optionsLayerLink = LayerLink();
   final OverlayPortalController _optionsViewController = OverlayPortalController(debugLabel: '_RawAutocompleteState');
 
+  // The box constraints that the field was last built with.
+  late BoxConstraints _fieldBoxConstraints;
```

Comment:
I'm surprised that no test caught this, but I think this will crash if the autocomplete widget is not in a layout builder. The `_fieldBoxConstraints` variable won't be assigned until the layout phase. 

As a workaround, putting the widget that takes that BoxConstraints in a layout builder will defer the evaluation of the late variable to the layout phase.

Overlay portal always lays out the `child` branch before the `overlayChild` branch so I think this should be safe (but I think the trick makes enough assumptions about layout builder's implementation and overlayportal's implementation, to merit throughout comments/testing)

Reply:
Here's my attempt: https://github.com/flutter/flutter/pull/143249/commits/49f98d6a00f9bae613c1c823f421177011c09519. Can you take a look at that and tell me if it's possible to get OverlayPortal to build `overlayChildBuilder` in the same frame as it builds `child`? I wasn't able to reproduce it, though I get what you're saying and it is worrying.

Reply:
Ah that probably doesn't work since the focus callback won't get called until the next microtask flush. If it's guaranteed that the options view won't show the same frame the field view shows, that should be safe (but `_fieldBoxConstraints` will still have a one frame delay).

Reply:
Ok got it, I've removed the LayoutBuilder, the test, and adjusted my comments. I will keep an eye out though in case anyone runs into this.

Reply:
I think the layout builder is probably still needed. If the text field gets a set of different input constraints, then this variable is going to change, but nothing is telling the menu to resize if it's already open?

Reply:
Well I guess OverlayPortal needs a new constructor that takes a `overlayChildLayoutBuilder` which gives you the size of the child before you build the overlay child.

Reply:
(Because the text field itself may change size, e.g., when there's an animation that resizes it)